## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#e59f38ab-2782-42b7-9636-a7f20ef1d4c1) |
| mint-gateway-azure.sh | :x: [more...](#d06bbdf3-1e84-4343-9086-e782fd7d7192) |
<a name="d06bbdf3-1e84-4343-9086-e782fd7d7192"></a>
### 5429-d08253e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:32607
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp ab99ae3aa053:/mint/log /tmp/mint-logs'
(1/11) Running aws-sdk-go tests ... done in 1 seconds
(2/11) Running aws-sdk-php tests ... done in 2 minutes and 12 seconds
(3/11) Running aws-sdk-ruby tests ... done in 52 seconds
(4/11) Running awscli tests ... done in 7 minutes and 25 seconds
(5/11) Running mc tests ... done in 6 minutes and 43 seconds
(6/11) Running minio-dotnet tests ... done in 1 minutes and 22 seconds
(7/11) Running minio-go tests ... done in 1 seconds
(8/11) Running minio-java tests ... done in 42 seconds
(9/11) Running minio-js tests ... done in 1 minutes and 56 seconds
(10/11) Running minio-py tests ... FAILED in 56 seconds
{
  "duration": 1219,
  "function": "list_incomplete_uploads(bucket_name, prefix, recursive)",
  "name": "minio-py:test_list_incomplete_uploads",
  "message": "Listed upload ids do not match the created ids",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1127, in test_list_incomplete_uploads\n    raise ValueError(\"Listed upload ids do not match the created ids\")\nValueError: Listed upload ids do not match the created ids\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1807, in main\n    test_list_incomplete_uploads(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1129, in test_list_incomplete_uploads\n    raise Exception(err)\nException: Listed upload ids do not match the created ids\n",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-a4f442a5-0dd2-4f7a-b907-79edacdc6371",
    "object_name": "939572e8-580d-4763-b8d5-f5f9ddc98863"
  }
}

Executed 9 out of 11 tests successfully.
```
<a name="e59f38ab-2782-42b7-9636-a7f20ef1d4c1"></a>
### 5429-d08253e/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30212
ACCESS_KEY:      AKIAJH7GQKYNU3C2ADXA
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp d8fb3a05095e:/mint/log /tmp/mint-logs'
(1/11) Running aws-sdk-go tests ... done in 4 seconds
(2/11) Running aws-sdk-php tests ... done in 1 minutes and 21 seconds
(3/11) Running aws-sdk-ruby tests ... done in 1 minutes and 20 seconds
(4/11) Running awscli tests ... done in 8 minutes and 8 seconds
(5/11) Running mc tests ... done in 6 minutes and 15 seconds
(6/11) Running minio-dotnet tests ... done in 1 minutes and 45 seconds
(7/11) Running minio-go tests ... done in 22 seconds
(8/11) Running minio-java tests ... FAILED in 53 seconds
{
  "name": "minio-java",
  "function": "setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
  "duration": 6463,
  "status": "FAIL",
  "error": "error occured\nErrorResponse(code=AccessDenied, message=Access Denied., bucketName=null, objectName=null, resource=/minio-java-test-16d7ps0/set-bucket-policy-read-write/minio-java-test-mkbspp, requestId=3L137, hostId=3L137)\nrequest={method=GET, url=http://minikube:30212/minio-java-test-16d7ps0/set-bucket-policy-read-write/minio-java-test-mkbspp, headers=Host: minikube:30212\nUser-Agent: Minio (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20180118T214815Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20180118/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Type: application/xml\nServer: Minio/DEVELOPMENT.2018-01-18T21-24-37Z (linux; amd64)\nVary: Origin\nX-Amz-Request-Id: 150B05D12AEA6130\nDate: Thu, 18 Jan 2018 21:48:15 GMT\nTransfer-Encoding: chunked\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1205), io.minio.MinioClient.executeGet(MinioClient.java:1298), io.minio.MinioClient.getObject(MinioClient.java:1606), io.minio.MinioClient.getObject(MinioClient.java:1510), FunctionalTest.testSetBucketPolicy(FunctionalTest.java:2427), FunctionalTest.setBucketPolicy_test4(FunctionalTest.java:2482), FunctionalTest.runQuickTests(FunctionalTest.java:2788), FunctionalTest.main(FunctionalTest.java:2830)]"
}

Executed 7 out of 11 tests successfully.
```

--- a/cmd/lockinfo-handlers_test.go
+++ b/cmd/lockinfo-handlers_test.go
@@ -18,27 +18,38 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 	"testing"
 	"time"
 )
 
 // TestListLocksInfo - Test for listLocksInfo.
 func TestListLocksInfo(t *testing.T) {
-	// Initialize globalNSMutex to validate listing of lock
-	// instrumentation information.
+	// reset global variables to start afresh.
+	resetTestGlobals()
+	// Initialize minio server config.
+	rootPath, err := newTestConfig(globalMinioDefaultRegion)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(rootPath)
+	// Initializing new XL objectLayer.
+	objAPI, _, xlErr := initTestXLObjLayer()
+	if xlErr != nil {
+		t.Fatalf("failed to init object layer")
+	}
+	// Make objLayer available to all internal services via globalObjectAPI.
+	globalObjLayerMutex.Lock()
+	globalObjectAPI = objAPI
+	globalObjLayerMutex.Unlock()
+	// Set globalIsXL to indicate that the setup uses an erasure code backend.
+	// initialize NSLock.
 	isDistXL := false
 	initNSLock(isDistXL)
-	objAPI := newObjectLayerFn()
-	if objAPI == nil {
-		t.Errorf("Failed to initialize object layer")
-	}
+
 	var nsMutex *nsLockMap
-	switch objAPI.(type) {
-	case *fsObjects:
-		nsMutex = objAPI.(*fsObjects).nsMutex
-	case *xlObjects:
-		nsMutex = objAPI.(*xlObjects).nsMutex
-	}
+
+	nsMutex = objAPI.(*xlObjects).nsMutex
 
 	// Acquire a few locks to populate lock instrumentation.
 	// Take 10 read locks on bucket1/prefix1/obj1

--- a/cmd/namespace-lock_test.go
+++ b/cmd/namespace-lock_test.go
@@ -39,6 +39,8 @@ func TestGetSource(t *testing.T) {
 
 // Tests functionality provided by namespace lock.
 func TestNamespaceLockTest(t *testing.T) {
+	isDistXL := false
+	initNSLock(isDistXL)
 	// List of test cases.
 	testCases := []struct {
 		lk               func(s1, s2, s3 string, t time.Duration) bool
@@ -155,7 +157,8 @@ func TestNamespaceLockTest(t *testing.T) {
 }
 
 func TestNamespaceLockTimedOut(t *testing.T) {
-
+	isDistXL := false
+	initNSLock(isDistXL)
 	// Get write lock
 	if !globalNSMutex.Lock("my-bucket", "my-object", "abc", 60*time.Second) {
 		t.Fatalf("Failed to acquire lock")
@@ -199,7 +202,8 @@ func TestNamespaceLockTimedOut(t *testing.T) {
 
 // Tests functionality to forcefully unlock locks.
 func TestNamespaceForceUnlockTest(t *testing.T) {
-
+	isDistXL := false
+	initNSLock(isDistXL)
 	// Create lock.
 	lock := globalNSMutex.NewNSLock("bucket", "object")
 	if lock.GetLock(newDynamicTimeout(60*time.Second, time.Second)) != nil {

